### DISCUSSION TITLE
imgui failed to build with "error LNK2019: unresolved external symbol ImGuiContext_cdecl ImGui::CreateContext(struct ImFontAtlas *)" PAUIDirect3DDevice9@@@Z)"

### DISCUSSION THREAD

**Discussion by YarIchkaT** [ocornut/imgui#8113] [New Users Build/Link/Run issues ONLY!]
### Version/Branch of Dear ImGui:

Version last

### Back-ends:

imgui_impl_win32.cpp + imgui_impl_dx9.cpp

### Compiler, OS:

Win 10 + VS 2022 

### Details:

**My Issue/Question:**

I've attached a screenshot of the error below, I'm trying to build, but I get about 25 errors.
Im use win62, but I made the project in 32 bits, Release x86

"error LNK2019: unresolved external symbol ImGuiContext_cdecl ImGui::CreateContext(struct ImFontAtlas *)" PAUIDirect3DDevice9@@@Z)"
Error file gui.obj

I have two gui files, one cpp and one header, it complains about lines of code in the cpp file.

I will also attach below the lines of code that give errors.

I could skip this problem and build a finished project, but it’s not mine, and I want to create something of my own, I want to learn how to do something myself, that’s why I want to solve the problem.

### Screenshots/Video:

![image](https://github.com/user-attachments/assets/72016555-e83a-40e3-a4a1-cb2429a0befd)

### Minimal, Complete and Verifiable Example code:

```cpp
// Absolutely ordinary lines of code
	ImGui::CreateContext();
	ImGui::StyleColorsDark();

// There are about 26 such ordinary lines of code that give an error
void gui::Destroy() noexcept
{
	ImGui_ImplDX9_Shutdown();
	ImGui_ImplWin32_Shutdown();
// Complains specifically about one bottom line
	ImGui::DestroyContext();

	SetWindowLongPtr(
		window,
		GWLP_WNDPROC,
		reinterpret_cast<LONG_PTR>(originalWindowProcess)
	);

	DestroyDirectX();
}

void gui::Render() noexcept
{
	ImGui_ImplDX9_NewFrame();
	ImGui_ImplWin32_NewFrame();
// Complains specifically about one bottom line
	ImGui::NewFrame();

// Complains specifically about the four bottom lines
	ImGui::Begin("ImGui Testing", &open);
	ImGui::End();

	ImGui::EndFrame();
	ImGui::Render();
	ImGui_ImplDX9_RenderDrawData(ImGui::GetDrawData());
}

// I don't see any point in writing all the code here, all the errors are identical and I gave several examples

```
// log file
  gui.cpp
  Generating Code...
gui.obj : error LNK2019: unresolved external symbol "struct ImGuiContext * __cdecl ImGui::CreateContext(struct ImFontAtlas *)" (?CreateContext@ImGui@@YAPAUImGuiContext@@PAUImFontAtlas@@@Z) referenced in function "void __cdecl gui::SetupMenu(struct IDirect3DDevice9 *)" (?SetupMenu@gui@@YAXPAUIDirect3DDevice9@@@Z)
gui.obj : error LNK2019: unresolved external symbol "void __cdecl ImGui::DestroyContext(struct ImGuiContext *)" (?DestroyContext@ImGui@@YAXPAUImGuiContext@@@Z) referenced in function "void __cdecl gui::Destroy(void)" (?Destroy@gui@@YAXXZ)
gui.obj : error LNK2019: unresolved external symbol "void __cdecl ImGui::NewFrame(void)" (?NewFrame@ImGui@@YAXXZ) referenced in function "void __cdecl gui::Render(void)" (?Render@gui@@YAXXZ)
gui.obj : error LNK2019: unresolved external symbol "void __cdecl ImGui::EndFrame(void)" (?EndFrame@ImGui@@YAXXZ) referenced in function "void __cdecl gui::Render(void)" (?Render@gui@@YAXXZ)
gui.obj : error LNK2019: unresolved external symbol "void __cdecl ImGui::Render(void)" (?Render@ImGui@@YAXXZ) referenced in function "void __cdecl gui::Render(void)" (?Render@gui@@YAXXZ)
gui.obj : error LNK2019: unresolved external symbol "struct ImDrawData * __cdecl ImGui::GetDrawData(void)" (?GetDrawData@ImGui@@YAPAUImDrawData@@XZ) referenced in function "void __cdecl gui::Render(void)" (?Render@gui@@YAXXZ)
gui.obj : error LNK2019: unresolved external symbol "void __cdecl ImGui::StyleColorsDark(struct ImGuiStyle *)" (?StyleColorsDark@ImGui@@YAXPAUImGuiStyle@@@Z) referenced in function "void __cdecl gui::SetupMenu(struct IDirect3DDevice9 *)" (?SetupMenu@gui@@YAXPAUIDirect3DDevice9@@@Z)
gui.obj : error LNK2019: unresolved external symbol "bool __cdecl ImGui::Begin(char const *,bool *,int)" (?Begin@ImGui@@YA_NPBDPA_NH@Z) referenced in function "void __cdecl gui::Render(void)" (?Render@gui@@YAXXZ)
gui.obj : error LNK2019: unresolved external symbol "void __cdecl ImGui::End(void)" (?End@ImGui@@YAXXZ) referenced in function "void __cdecl gui::Render(void)" (?Render@gui@@YAXXZ)
gui.obj : error LNK2019: unresolved external symbol "bool __cdecl ImGui_ImplWin32_Init(void *)" (?ImGui_ImplWin32_Init@@YA_NPAX@Z) referenced in function "void __cdecl gui::SetupMenu(struct IDirect3DDevice9 *)" (?SetupMenu@gui@@YAXPAUIDirect3DDevice9@@@Z)
gui.obj : error LNK2019: unresolved external symbol "void __cdecl ImGui_ImplWin32_Shutdown(void)" (?ImGui_ImplWin32_Shutdown@@YAXXZ) referenced in function "void __cdecl gui::Destroy(void)" (?Destroy@gui@@YAXXZ)
gui.obj : error LNK2019: unresolved external symbol "void __cdecl ImGui_ImplWin32_NewFrame(void)" (?ImGui_ImplWin32_NewFrame@@YAXXZ) referenced in function "void __cdecl gui::Render(void)" (?Render@gui@@YAXXZ)
gui.obj : error LNK2019: unresolved external symbol "bool __cdecl ImGui_ImplDX9_Init(struct IDirect3DDevice9 *)" (?ImGui_ImplDX9_Init@@YA_NPAUIDirect3DDevice9@@@Z) referenced in function "void __cdecl gui::SetupMenu(struct IDirect3DDevice9 *)" (?SetupMenu@gui@@YAXPAUIDirect3DDevice9@@@Z)
gui.obj : error LNK2019: unresolved external symbol "void __cdecl ImGui_ImplDX9_Shutdown(void)" (?ImGui_ImplDX9_Shutdown@@YAXXZ) referenced in function "void __cdecl gui::Destroy(void)" (?Destroy@gui@@YAXXZ)
gui.obj : error LNK2019: unresolved external symbol "void __cdecl ImGui_ImplDX9_NewFrame(void)" (?ImGui_ImplDX9_NewFrame@@YAXXZ) referenced in function "void __cdecl gui::Render(void)" (?Render@gui@@YAXXZ)
gui.obj : error LNK2019: unresolved external symbol "void __cdecl ImGui_ImplDX9_RenderDrawData(struct ImDrawData *)" (?ImGui_ImplDX9_RenderDrawData@@YAXPAUImDrawData@@@Z) referenced in function "void __cdecl gui::Render(void)" (?Render@gui@@YAXXZ)
gui.obj : error LNK2019: unresolved external symbol "long __cdecl ImGui_ImplWin32_WndProcHandler(struct HWND__ *,unsigned int,unsigned int,long)" (?ImGui_ImplWin32_WndProcHandler@@YAJPAUHWND__@@IIJ@Z) referenced in function "long __stdcall WindowProcess(struct HWND__ *,unsigned int,unsigned int,long)" (?WindowProcess@@YGJPAUHWND__@@IIJ@Z)

**Comment by Macddee** [comment]
im also having the exact errors and im certain i  have included all the libs and headers but nothin g seemes to be working. have you found any help yet?
